### ISSUE TITLE
Run Multiple R CMD Check

### ISSUE THREAD

**Issue by muschellij2** [ANTsX/ANTsRCore#169]
I think Ubuntu release is the only testing being done - can we add a GH action for multiple OSes?

**Comment by ncullen93** [comment]
Absolutely, I can try to add some. I wonder what the state of windows support is... Anyways, I've started a PR here https://github.com/ANTsX/ANTsRCore/pull/170. Feel free to comment on it, edit the branch, or create your own if you want.

**Comment by cookpa** [comment]
Just seeing this - Mac runners are having some issues right now but can add Mac when they are working again

Windows builds will not work until we can get ITKR building. Rtools wants to use GCC 13, and I could not convince it to use an earlier compiler. So it kept failing to build ITK because of this error

https://discourse.itk.org/t/building-itk-error-with-algorithim-module/5927
